crates/astria-composer/src/executor/mod.rs

```diff
@@ -212,25 +214,26 @@ impl Executor {
     ///
     /// # Errors
     /// An error is returned if connecting to the sequencer fails.
-    // Allow: Refactored in <insert GH here>. Current logging events will show as errors in otel,
-    // refactor addresses this as well.
-    #[allow(clippy::too_many_lines)]
     pub(super) async fn run_until_stopped(mut self) -> eyre::Result<()> {
         select!(
             biased;
             () = self.shutdown_token.cancelled() => {
-                info!("received shutdown signal while running initialization routines; exiting");
-                return Ok(());
+                log_exit(&Ok("received shutdown signal while running initialization routines; exiting"));
+                return Ok(())
             }
 
             res = self.pre_run_checks() => {
+                log_exit(&res);
                 res.wrap_err("required pre-run checks failed")?;
             }
         );
         let mut submission_fut: Fuse<Instrumented<SubmitFut>> = Fuse::terminated();
         let mut nonce = get_latest_nonce(self.sequencer_client.clone(), self.address, self.metrics)
             .await
-            .wrap_err("failed getting initial nonce from sequencer")?;
+            .map_err(|err| {
+                log_exit(&Err(eyre!(err.to_string())));
```

Comment:
Why turn this into a string? You can do:

```rust
get_latest_nonce().wrap_err("failed getting initial nonce from sequencer").inspect_err(|err| report_exit(err))
```

You will also need to redefine `report_exit` to be something like `fn report_exit(reason: Result<&str, &eyre::Report>)` - but then you can make use of `Result::as_ref` where necessary.